lib/sailor.js

```diff
@@ -322,29 +346,118 @@ class Sailor {
                 const headers = _.clone(outgoingMessageHeaders);
                 headers.messageId = data.id || headers.messageId;
                 logger.trace({
-                    messagesCount: self.messagesCount,
+                    messagesCount: that.messagesCount,
                     messageProcessingTime: Date.now() - timeStart
                 }, 'processMessage emit data');
 
                 headers.end = new Date().getTime();
 
                 if (stepData.is_passthrough === true) {
+                    data.passthrough = { ...origPassthrough };
                     if (settings.NO_SELF_PASSTRHOUGH) {
                         const { stepId } = incomingMessageHeaders;
                         if (stepId) {
                             data.passthrough = Object.assign({}, origPassthrough, {
                                 [stepId]: Object.assign({}, _.omit(payload, 'passthrough'))
                             });
                         }
+                    }
+                }
+
+                data.headers = data.headers || {};
+                const { body, passthrough } = data;
+
+                if (settings.EMIT_LIGHTWEIGHT_MESSAGE) {
+                    logger.info('Outgoing lightweight is enabled, going to check size.');
+                    const bodyBuf = Buffer.from(JSON.stringify(body), 'utf-8');
+                    const passthroughBufs = Object.keys(passthrough).map(stepId => ({
```

Comment:
Correct me if I'm wrong, but passthrough might already be in maester. 
So sometimes there are no need to do all this. for passthrough

Reply:
yep, and actual upload does not happen if incoming passthrough has and id header

Reply:
here https://github.com/elasticio/sailor-nodejs/pull/150/files#diff-49130a3beafac89ab92d914a0be244a5R393